### PR TITLE
Add regional stats view

### PR DESCRIPTION
**Pull request by Levia**
## Description

Adds  regional statistics view with data calculated starting from country statistics.
Add also a view necessary to fetch and manipulate data for the aichi 11 target dashboard.

### CODE REVIEW

File: views/regional_statistics_view/20190913123830.sql [hunk 1]
```diff
@@ -0,0 +1,43 @@
+SELECT region_id, land_area, eez_area, ts_area,
+  (COALESCE(pa_land_area,0) + COALESCE(pa_marine_area,0)) AS pa_area,
+  pa_land_area, pa_marine_area, pa_eez_area, pa_ts_area,
+  ((COALESCE(pa_land_area,0) + COALESCE(pa_marine_area,0)) /
+    (land_area + COALESCE(eez_area, 0) + COALESCE(ts_area,0))*100) AS percentage_pa_cover,
+  (COALESCE(pa_land_area,0) / land_area * 100) AS percentage_pa_land_cover,
+  (COALESCE(pa_marine_area,0) / marine_area * 100) AS percentage_pa_marine_cover,
+  (COALESCE(well_connected_area,0) / (land_area + marine_area) * 100) AS percentage_well_connected,
+  (COALESCE(importance_area,0) / (land_area + marine_area) * 100) AS percentage_importance,
+  (COALESCE(pame_pa_land_area,0) / COALESCE(pa_land_area,0) * 100) AS pame_percentage_pa_land_cover,
```
Comment (lucacug) [inline views/regional_statistics_view/20190913123830.sql:10]: I will check for the divisions by zero, not sure if SQL complains in the same way as Rails does...

File: views/regional_statistics_view/20190916102330.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
+SELECT region_id, land_area, eez_area, ts_area,
+  (COALESCE(pa_land_area,0) + COALESCE(pa_marine_area,0)) AS pa_area,
+  pa_land_area, pa_marine_area, pa_eez_area, pa_ts_area, jrc_regional_area,
+  ((COALESCE(pa_land_area,0) + COALESCE(pa_marine_area,0)) /
+    (land_area + COALESCE(eez_area, 0) + COALESCE(ts_area,0))*100) AS percentage_pa_cover,
+  (COALESCE(pa_land_area,0) / land_area * 100) AS percentage_pa_land_cover,
+  (COALESCE(pa_marine_area,0) / marine_area * 100) AS percentage_pa_marine_cover,
+  (COALESCE(well_connected_area,0) / jrc_regional_area * 100) AS percentage_well_connected,
+  (COALESCE(importance_area,0) / jrc_regional_area * 100) AS percentage_importance,
+  (COALESCE(pame_pa_land_area,0) / COALESCE(pa_land_area,0) * 100) AS pame_percentage_pa_land_cover,
```
Comment (lucacug) [inline views/regional_statistics_view/20190916102330.sql:10]: same as above

Comment (Levia) [reply views/regional_statistics_view/20190916102330.sql:10]: Good point. I'm thinking about this as, if any of that is really 0, that means that something is not quite right with the data and the result won't probably be acceptable.

**Comment by Levia** [comment]
@lucacug I went for NULLIF so that the division will return NULL instead of throwing a division by zero error.
When a resulting value is NULL instead of 0 or a number, it's clear that something with the data is wrong.

**Pull request merged**